src/NServiceBus.Core.Tests/AssemblyScanner/AssemblyScannerTests.cs

```diff
@@ -37,32 +41,226 @@ public void Non_system_assemblies_should_be_included()
+                builder.AppendLine($"[assembly: AssemblyFileVersion(\"{version}\")]");
+
+                builder.AppendFormat("namespace {0} {{ ", Namespace);
+
+                var provider = new CSharpCodeProvider();
+                var param = new CompilerParameters(new string[] { }, nameWithExtension);
+                param.GenerateExecutable = false;
+                param.OutputAssembly = FilePath = Path.Combine(GetTestAssemblyDirectory(), FileName);
+                param.TempFiles = new TempFileCollection(GetTestAssemblyDirectory(), false);
+
+                foreach (var reference in references)
+                {
+                    builder.AppendLine($"using {reference.Namespace};");
+                    param.ReferencedAssemblies.Add(reference.Assembly.Location);
+                }
+
+
+                builder.AppendLine("public class Foo { public Foo() {");
+                foreach (var reference in references)
+                {
+                    builder.AppendLine($"new {reference.Namespace}.Foo();");
+                }
+                builder.AppendLine("} } }");
+
+                var result = provider.CompileAssemblyFromSource(param, builder.ToString());
+                ThrowIfCompilationWasNotSuccessful(result);
+                Assembly = result.CompiledAssembly;
+            }
+
+            static string GetTestAssemblyDirectory()
+            {
+                var directoryName = GetAssemblyDirectory();
+                var directory = Path.Combine(directoryName, "assemblyscannerfiles");
+                if (!Directory.Exists(directory))
+                {
+                    Directory.CreateDirectory(directory);
+                }
+                return directory;
+            }
+
+            static string GetAssemblyDirectory()
+            {
+                var codeBase = Assembly.GetExecutingAssembly().CodeBase;
+                var uri = new UriBuilder(codeBase);
+                var path = Uri.UnescapeDataString(uri.Path);
+                return Path.GetDirectoryName(path);
+            }
+
+            static void ThrowIfCompilationWasNotSuccessful(CompilerResults results)
+            {
+                if (results.Errors.HasErrors)
+                {
+                    var errors = new StringBuilder("Compiler Errors :\r\n");
+                    foreach (CompilerError error in results.Errors)
+                    {
+                        errors.AppendFormat("Line {0},{1}\t: {2}\n",
+                            error.Line, error.Column, error.ErrorText);
+                    }
+                    throw new Exception(errors.ToString());
+                }
+            }
+
+            public static implicit operator Assembly(DynamicAssembly dynamicAssembly)
+            {
+                return dynamicAssembly.Assembly;
+            }
+
+            public void Dispose()
+            {
+                File.Delete(FilePath);
```

Comment:
When I run these tests locally, I'm running into a problem where the delete is failing with an UnauthorizedException. Do we need to worry about cleaning up these files?